File: app/src/main/java/at/tomtasche/reader/ui/widget/ProgressDialogFragment.java
package at.tomtasche.reader.ui.widget;

import android.annotation.SuppressLint;
import android.app.Dialog;
import android.app.ProgressDialog;
import android.os.Bundle;

import androidx.fragment.app.DialogFragment;
import at.tomtasche.reader.R;

@SuppressLint("ValidFragment")
public class ProgressDialogFragment extends DialogFragment {

    public static final String FRAGMENT_TAG = "progress_dialog";

    private ProgressDialog progressDialog;
    private boolean isUpload;

    public ProgressDialogFragment() {
        super();
    }

    public ProgressDialogFragment(boolean isUpload) {
        this();

        this.isUpload = isUpload;
    }

    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        progressDialog = new ProgressDialog(getActivity());

        int title;
        if (isUpload) {
            title = R.string.dialog_uploading_title;
        } else {
            title = R.string.dialog_loading_title;
        }

        progressDialog.setTitle(getString(title));

        String message = getString(R.string.dialog_generic_loading_message);
        if (isUpload) {
            message += " " + getString(R.string.dialog_uploading_message_appendix);
        }

        progressDialog.setMessage(message);
        progressDialog.setIndeterminate(true);

        // known issue that causes infinite progressdialog
        progressDialog.setCancelable(true);
        setCancelable(true);

        return progressDialog;
    }
}

File: app/src/main/java/app/opendocument/droid/ui/widget/ProgressDialogFragment.kt
package app.opendocument.droid.ui.widget

import android.app.Dialog
import android.os.Bundle
import android.widget.TextView
import androidx.fragment.app.DialogFragment
import app.opendocument.droid.R
import com.google.android.material.dialog.MaterialAlertDialogBuilder

/**
 * Built on MaterialAlertDialogBuilder rather than the framework ProgressDialog, which resolves
 * android:alertDialogTheme and so would have stayed light once the app theme became DayNight.
 */
class ProgressDialogFragment : DialogFragment() {

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val view = layoutInflater.inflate(R.layout.dialog_progress, null)
        view.findViewById<TextView>(R.id.progress_message).text =
            getString(R.string.dialog_generic_loading_message)

        // known issue that causes infinite progressdialog
        setCancelable(true)

        return MaterialAlertDialogBuilder(requireContext())
            .setTitle(R.string.dialog_loading_title)
            .setView(view)
            .setCancelable(true)
            .create()
    }

    companion object {
        const val FRAGMENT_TAG: String = "progress_dialog"
    }
}
